tensorboard/webapp/alert/types.ts

```diff
@@ -0,0 +1,25 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+/**
+ * An alert structure used when creating newly reported alerts.
+ */
+export interface AlertReport {
+  details: string;
```

Comment:
I do not think this is sufficient to migrate existing code. We will need to have an optional actionType for an alert that can have actions (you may also want to have localized string for those actions).